tests/SpecTests/ResultExpectation.php

```diff
@@ -72,6 +72,19 @@ public static function fromChangeStreams(stdClass $result, callable $assertionCa
         return $o;
     }
 
+    public static function fromCrud(stdClass $operation, $defaultAssertionType)
+    {
+        if (property_exists($operation, 'result') && !self::isErrorResult($operation->result)) {
+            $assertionType = $operation->result === null ? self::ASSERT_NULL : $defaultAssertionType;
+            $expectedValue = $operation->result;
+        } else {
+            $assertionType = self::ASSERT_NOTHING;
+            $expectedValue = null;
+        }
+
+        return new self($assertionType, $expectedValue);
+    }
```

Comment:
I believe this is identical to `fromTransactions()`. I think both methods should stay, but maybe we should extra the body into a new function so we can do something like:

```
list($assertionType, $expectedValue) = getAssertionTypeAndExpectedValue($operation);
```

Ideally, retryable writes will ultimately be migrated to use consistent syntax. I'm not sure if retryable reads (coming soon) does, too, so this refactoring could certainly wait until we have 3+ duplications.

Reply:
I'll refactor the `fromCrud`, `fromRetryableWrites` and `fromTransactions` methods to add a `getAssertion...` method like you suggested. We can then always keep more significant exceptions to the norm like `fromChangeStreams` while avoiding duplicating the usual checks. 👍

Reply:
So, after starting to refactor I decided to put this off for now. The `fromRetryableWrites` helper is different, which means that I'll most likely do once we add retryable reads.